src/manager.cpp

```diff
@@ -461,8 +461,27 @@ void Manager::collectSingleFruVpd(
 void Manager::deleteSingleFruVpd(
     const sdbusplus::message::object_path& i_dbusObjPath)
 {
-    // Dummy code to supress unused variable warning. To be removed.
-    logging::logMessage(std::string(i_dbusObjPath));
+    try
+    {
+        if (std::string(i_dbusObjPath).empty())
+        {
+            throw std::runtime_error(
+                "Given input is empty. Cannot perform deletion for: " +
+                std::string(i_dbusObjPath));
+        }
+
+        if (m_worker.get() == nullptr)
+        {
+            throw std::runtime_error("Failed to delete VPD for: " +
+                                     std::string(i_dbusObjPath));
+        }
+
+        m_worker->deleteFruVpd(std::string(i_dbusObjPath));
```

Comment:
why are you moving the implementation to worker class?

Reply:
@SunnySrivastava1984 asked to move the implementation to worker class.